### PR TITLE
Add note about devhub/latest tag

### PR DESCRIPTION
**Pull request by sacha-l**
Adding a note about our internal process for keeping our node template, tutorial write-ups and solutions up-to-date.

Not exactly sure where this should live but this feels ok for now. 

CC: @NukeManDan: step 3. needs a little more details. Can you add your edits in please? Just commit directly to this PR 😎 .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -187,6 +187,22 @@ production as a docker image and push it to Docker hub at
...
 
+## Process for using `devhub/latest` and `latest` tags on templates
+
+The devhub team uses the Node Template for tutorials and guides which requires making sure that both the templates and the write-ups (including solutions) are always up to date.
+We have a system that ensures that content in our documentation hub always builds correctly by using the `devhub/latest` tag for any Substrate crate. 
+
+This requires following a few steps for every new release of Substrate upstream:
+
+1. Go to the [release tags in the Substrate repository](https://github.com/paritytech/substrate/tags).
+1. Set the `devhub/latest` tag to the latest monthly release.
```
Comment (NukeManDan) [inline README.md:198]:
```suggestion
1. Set the `devhub/latest` tag to the latest [Polkadot release](https://github.com/paritytech/polkadot/releases) (`polakdot-v<Major>.<Minor>.<Patch>`, not the release candidates apppended `...-rc#`).
```

File: README.md [hunk 2]
```diff
@@ -187,6 +187,22 @@ production as a docker image and push it to Docker hub at
 The image is then launched as a service in the next CI/CD job. `checklinks` is run against the
 running gatsby site in the docker container to check all links.
 
+## Process for using `devhub/latest` and `latest` tags on templates
+
+The devhub team uses the Node Template for tutorials and guides which requires making sure that both the templates and the write-ups (including solutions) are always up to date.
```
Comment (NukeManDan) [inline README.md:192]: Lisa said style for template's is not a proper noun... I think we need this in the style guide and a follow up PR to make uniform. cc @lisa-parity

File: README.md [hunk 3]
```diff
@@ -187,6 +187,22 @@ production as a docker image and push it to Docker hub at
...
+We have a system that ensures that content in our documentation hub always builds correctly by using the `devhub/latest` tag for any Substrate crate. 
+
+This requires following a few steps for every new release of Substrate upstream:
+
+1. Go to the [release tags in the Substrate repository](https://github.com/paritytech/substrate/tags).
+1. Set the `devhub/latest` tag to the latest monthly release.
+1. Create a new node template release for the latest monthly (e.g. `monthly-2021-12`). This assumes following the steps to [generate the new template](https://github.com/paritytech/substrate-template-generator).
```
Comment (NukeManDan) [inline README.md:199]:
```suggestion
1. Create a new node template release for the latest monthly (e.g. `polkadot-v-0.9.13`). This assumes following the steps to [generate the new template](https://github.com/paritytech/substrate-template-generator).
```

File: README.md [hunk 4]
```diff
@@ -187,6 +187,22 @@ production as a docker image and push it to Docker hub at
...
+1. Then go to the [release tags in the node template repository](https://github.com/substrate-developer-hub/substrate-node-template/tags).
+1. Make the node template's `latest` tag point to the newly generated node template (i.e. in the newly generated node template directory, `git tag latest && git push`).
```
Comment (NukeManDan) [inline README.md:201]:
```suggestion
1. Make the node template's `latest` tag point to the newly generated node template:
   ```bash
   # We want to tag the approprite commit, this is very likely now main
   # as you just merged a PR into `main` with any changes required from upstream
   cd substrate-node-template
   git checkout main 
   git pull
   # Move the `latest` tag locally from it's prior location 
   git tag -d latest && git tag latest
   # We need to force this tag change upstream **for the template!**
   git push origin latest --force.
    ```
```

File: README.md [hunk 5]
```diff
@@ -187,6 +187,22 @@ production as a docker image and push it to Docker hub at
...
+
+The result is that only the solutions to tutotials need to be checked. They could either remain on a previous release, in which case the chore would be to update all `Cargo.toml` files with the same tag. Or they could remain the same however, the chore would be to ensure the solution builds, fixing any upstream breaking changes.
```
Comment (NukeManDan) [inline README.md:203]:
```suggestion
The result is that only the solutions to tutorials need to be checked.
They could either remain on a previous release, in which case the chore would be to update all `Cargo.toml` files with the same tag.
Or they could remain the same however, the chore would be to ensure the solution builds, fixing any upstream breaking changes.
```

File: README.md [hunk 6]
```diff
@@ -187,6 +187,22 @@ production as a docker image and push it to Docker hub at
...
+
+This process also applies for the front-end template and is being tested for other templates such as Frontier and Cumulus.
```
Comment (NukeManDan) [inline README.md:205]: Just not using the template generator... this is all manual still.

**Review by NukeManDan — CHANGES_REQUESTED** [review]
Good start! :grinning: 

I think we should make these changes, and also describe the E2E testing required *before* anyone moves this tag. You make changes that make it into `main` on the templates, and *only after testing* do you move the latest tag. So the steps are in need of being tweaked.

File: README.md (hunk 2 above)
Comment (sacha-l) [reply README.md:192]:
```suggestion
The devhub team uses the node template for tutorials and guides which requires making sure that both the templates and the write-ups (including solutions) are always up to date.
```

File: README.md (hunk 6 above)
Comment (sacha-l) [inline README.md:218]:
```suggestion
This process also applies for the front-end template and is being tested for other templates such as Frontier and Cumulus. 
However, for templates other than the node template, the process is manual (i.e. it doesn't use the template generator mentioned above).
```

**Review by sacha-l — COMMENTED** [review]
Thanks for the review 😃

File: README.md (hunk 6 above)
Comment (sacha-l) [reply README.md:218]: Included in commit.

File: README.md [hunk 7]
```diff
@@ -187,6 +187,36 @@ production as a docker image and push it to Docker hub at
...
 
+## Process for using `devhub/latest` and `latest` tags on templates
+
+The devhub team uses the node template for tutorials and guides which requires making sure that both the templates and the write-ups (including solutions) are always up to date.
+We have a system that ensures that content in our documentation hub always builds correctly by using the `devhub/latest` tag for any Substrate crate. 
+
+This requires following a few steps for every new release of Substrate upstream:
+
+1. Go to the [release tags in the Substrate repository](https://github.com/paritytech/substrate/tags).
+1. Set the `devhub/latest` tag to the latest [Polkadot release](https://github.com/paritytech/polkadot/releases) (`polakdot-v<Major>.<Minor>.<Patch>`, not the release candidates apppended `...-rc#`).
```
Comment (sacha-l) [inline README.md:198]: @NukeManDan : FYI we used Substrate's latest [here](https://github.com/paritytech/substrate/releases/tag/devhub%2Flatest).

**Comment by sacha-l** [comment]
> Good start! 😀
> 
> I think we should make these changes, and also describe the E2E testing required _before_ anyone moves this tag. You make changes that make it into `main` on the templates, and _only after testing_ do you move the latest tag. So the steps are in need of being tweaked.

Do you have any other suggestions to address the E2E testing? Feel free to commit directly to this branch.

**Comment by sacha-l** [comment]
@NukeManDan : ping ! :)

**Comment by sacha-l** [comment]
This can be closed I believe. We're no longer following this process anymore. @NukeManDan can you find a reference PR or issue to add more context and close?

**Comment by lisa-parity** [comment]
This seems to be obsolete with the other changes and this tag never made it into the core substrate repo, so I suggest closing this as clutter that won't be merged.

**Pull request merged**